pysat/instruments/methods/icon.py

```diff
@@ -0,0 +1,330 @@
+    else:
+        eyr, edoy = pysat.utils.time.getyrdoy(stop)
+        eyrdoy = eyr * 1000 + edoy
+    # apply date filter
+    idx, = np.where((yrdoys >= syrdoy) & (yrdoys <= eyrdoy))
+    remote_days = np.array(remote_days)[idx].tolist()
+
+    # leading path, get any directory elements between the year/doy
+    # and the start of actual files on SSL server
+    leading_dirs = remote_fname.split('/')
+    leading_dirs = leading_dirs[:-1]
+    leading_dirs = '/'.join(leading_dirs)
+    if leading_dirs != '':
+        leading_dirs += '/'
+
+    # get a list of files now that all leading portions determined
+    day_file_list = []
+    for remote_day in remote_days:
+        temp_dirs = []
+        ftp.cwd(remote_day + '/' + leading_dirs)
+        ftp.dir(temp_dirs.append)
+        for item in temp_dirs:
+            # parse raw string
+            parsed = item.split(' ')
+            day_file_list.append(remote_day + '/' + leading_dirs + parsed[-1])
+
+    # we now have a list of all files in the instrument data directories
+    # need to whittle list down to the versions and revisions most appropriate
+    search_dict = pysat._files.construct_searchstring_from_format(remote_fname)
+    search_str = '*/' + search_dict['search_string']
+    remote_files = fnmatch.filter(day_file_list, search_str)
+
+    # pull out date information from the files
+    stored = pfiles.parse_fixed_width_filenames(remote_files, remote_fname)
+    output = pfiles.process_parsed_filenames(stored)
+    # return information, limited to start and stop dates
+    return output[start:stop]
+
+
+def ssl_download(date_array, tag, sat_id, data_path=None,
+                 user=None, password=None, supported_tags=None):
+    """Download ICON data from public area of SSL ftp server
+
+    Parameters
+    ----------
+    date_array : array-like
+        list of datetimes to download data for. The sequence of dates need not
+        be contiguous.
+    tag : string ('')
+        Tag identifier used for particular dataset. This input is provided by
+        pysat.
+    sat_id : string  ('')
+        Satellite ID string identifier used for particular dataset. This input
+        is provided by pysat.
+    data_path : string (None)
+        Path to directory to download data to.
+    user : string (None)
+        User string input used for download. Provided by user and passed via
+        pysat. If an account is required for downloads this routine here must
+        error if user not supplied.
+    password : string (None)
+        Password for data download.
+    **kwargs : dict
+        Additional keywords supplied by user when invoking the download
+        routine attached to a pysat.Instrument object are passed to this
+        routine via kwargs.
+
+    Returns
+    --------
+    Void : (NoneType)
+        Downloads data to disk.
+
+    """
+
+    try:
+        ftp_dict = supported_tags[sat_id][tag]
+    except KeyError:
+        raise ValueError('sat_id/tag name unknown.')
```

Comment:
`ftp_dict` is not used in this function.  Since we're getting the list of files below, do we need it here?

Reply:
No, that is leftover code. The download method went through a few revisions. Thanks for noticing.

Reply:
Don’t thank me, thank flake8. 😸